http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/RepeatableInputStreamRequestEntity.java

```diff
@@ -78,17 +79,17 @@ public RepeatableInputStreamRequestEntity(final SdkHttpFullRequest request) {
          * buffer the entire stream contents into memory to determine
          * the content length.
          */
-        long contentLength = request.getFirstHeaderValue("Content-Length")
-                .map(this::parseContentLength)
-                .orElse(-1L);
+        long contentLength = SdkHttpUtils.firstMatchingHeader(request.headers(), "Content-Length")
```

Comment:
I moved it to a utility method because it felt useful even if we don't have access to an HTTP request/response. For example, code that is using the builder or HTTP plugin code before it has created the request/response can use the functionality now.

The reason I didn't add a convenience method onto both the builder and request/response objects to delegate to the util class:
- There were already a lot of other methods on the builder/request already.
- There are a few things in SdkHttpUtils that could make sense on the interfaces, and I was worried about the subjectivity of "should we add a convenience method for this to the builder/request interfaces?" Granted, I did add `getUri` to the request, so I'm being a bit hypocritical here. That was to avoid the utils -> http-spi dependency.

Let me know if you feel like it's worth keeping.

Reply:
I dunno, I feel like the interfaces should be rich enough and useful enough for us and SPI implementors work with. +1 for the utility but I think it would also be nice on the interface.